### ISSUE TITLE
Bluetooth is refusing to work

### ISSUE THREAD

**Issue by michaeljhkim** [DanielOgorchock/linux#45]
So the USB system works 100% fine, but trying to get my pro controller to work with bluetooth is impossible. 

It says that it has connected through bluetooth, but it is not recognized as an input device of any kind. My controller works if I open up steam, but steam uses it's own input driver system of course.

I can confidentially say that hid_nintendo has a problem with bluetooth.

EDIT: I realized that I should specify that I am using the gullikit king kong pro 2 controller in switch mode

**Comment by michaeljhkim** [comment]
here is the dmesg:

[109418.070018] nintendo 0005:057E:2009.0048: hidraw8: BLUETOOTH HID v80.01 Gamepad [Pro Controller] on d4:54:8b:d8:74:2f
[109422.135818] nintendo 0005:057E:2009.0048: failed reading SPI flash; ret=-110
[109422.135821] nintendo 0005:057E:2009.0048: using factory cal for left stick
[109424.162460] nintendo 0005:057E:2009.0048: failed reading SPI flash; ret=-110
[109424.162465] nintendo 0005:057E:2009.0048: using factory cal for right stick
[109426.192429] nintendo 0005:057E:2009.0048: failed reading SPI flash; ret=-110
[109426.192445] nintendo 0005:057E:2009.0048: Failed to read left stick cal, using defaults; e=-110
[109428.215795] nintendo 0005:057E:2009.0048: failed reading SPI flash; ret=-110
[109428.215808] nintendo 0005:057E:2009.0048: Failed to read right stick cal, using defaults; e=-110
[109430.242371] nintendo 0005:057E:2009.0048: failed reading SPI flash; ret=-110
[109430.242387] nintendo 0005:057E:2009.0048: using factory cal for IMU
[109432.269006] nintendo 0005:057E:2009.0048: failed reading SPI flash; ret=-110
[109432.269024] nintendo 0005:057E:2009.0048: Failed to read IMU cal, using defaults; ret=-110
[109432.269027] nintendo 0005:057E:2009.0048: Unable to read IMU calibration data
[109434.295778] nintendo 0005:057E:2009.0048: Failed to set report mode; ret=-110
[109434.296079] nintendo 0005:057E:2009.0048: probe - fail = -110
[109434.296086] nintendo: probe of 0005:057E:2009.0048 failed with error -110

**Comment by EuanFH** [comment]
yeah I'm getting the exact same issue. looking at the driver it seems to be timing out waiting for a sub command to run.

**Comment by zenmaster3000** [comment]
I'm also using an non-official nintendo controller (8bitdo Pro 2 Controller)
```
[Sat Aug 12 02:54:17 2023] nintendo 0005:057E:2009.0009: unknown main item tag 0x0
[Sat Aug 12 02:54:17 2023] nintendo 0005:057E:2009.0009: hidraw4: BLUETOOTH HID v80.01 Gamepad [Pro Controller] on 44:af:28:13:a0:91
[Sat Aug 12 02:54:21 2023] nintendo 0005:057E:2009.0009: failed reading SPI flash; ret=-110
[Sat Aug 12 02:54:21 2023] nintendo 0005:057E:2009.0009: using factory cal for left stick
[Sat Aug 12 02:54:23 2023] nintendo 0005:057E:2009.0009: failed reading SPI flash; ret=-110
[Sat Aug 12 02:54:23 2023] nintendo 0005:057E:2009.0009: using factory cal for right stick
[Sat Aug 12 02:54:25 2023] nintendo 0005:057E:2009.0009: failed reading SPI flash; ret=-110
[Sat Aug 12 02:54:25 2023] nintendo 0005:057E:2009.0009: Failed to read left stick cal, using defaults; e=-110
[Sat Aug 12 02:54:27 2023] nintendo 0005:057E:2009.0009: failed reading SPI flash; ret=-110
[Sat Aug 12 02:54:27 2023] nintendo 0005:057E:2009.0009: Failed to read right stick cal, using defaults; e=-110
[Sat Aug 12 02:54:29 2023] nintendo 0005:057E:2009.0009: failed reading SPI flash; ret=-110
[Sat Aug 12 02:54:29 2023] nintendo 0005:057E:2009.0009: using factory cal for IMU
[Sat Aug 12 02:54:31 2023] nintendo 0005:057E:2009.0009: failed reading SPI flash; ret=-110
[Sat Aug 12 02:54:31 2023] nintendo 0005:057E:2009.0009: Failed to read IMU cal, using defaults; ret=-110
[Sat Aug 12 02:54:31 2023] nintendo 0005:057E:2009.0009: Unable to read IMU calibration data
[Sat Aug 12 02:54:33 2023] nintendo 0005:057E:2009.0009: Failed to set report mode; ret=-110
[Sat Aug 12 02:54:33 2023] nintendo 0005:057E:2009.0009: probe - fail = -110
[Sat Aug 12 02:54:33 2023] nintendo: probe of 0005:057E:2009.0009 failed with error -110
```

**Comment by zenmaster3000** [comment]
So it seems our problems are a duplicate of another issue here, there is a workaround:

https://github.com/DanielOgorchock/linux/issues/10#issuecomment-1321074341

but I have not tried it myself.

**Comment by EuanFH** [comment]
No this is a separate issue I'm pretty sure the vendor id and product id is correct for the king long 2 controller it copies Nintendo's official ones.

**Comment by zenmaster3000** [comment]
Could you post then your dmesg?
